src/wagtail_factories/blocks.py

```diff
@@ -56,15 +51,64 @@ def generate(self, step, params):
                     continue
 
                 index = int(index)
-                result[index][block_name][param] = value
+                mappings[index][block_name][param] = value
+        return mappings
 
-        retval = []
-        for index, block_items in sorted(result.items()):
+    @classmethod
+    def _generate(cls, strategy, params):
+        indexed_mappings = cls._get_indexed_mappings(params)
+        stream_data = []
+        for index, block_items in sorted(indexed_mappings.items()):
             for block_name, block_params in block_items.items():
-                block_factory = self.get_factory_for_block(block_name)
-                value = block_factory(**block_params)
-                retval.append((block_name, value))
-        return retval
+                block_factory = cls._get_block_factory(block_name)
+                if isinstance(block_factory, ListBlockFactory):
+                    stream_data.append((block_name, block_factory(**block_params)))
```

Comment:
@bcdickinson Tests are passing but I think we need to pass the strategy through to ListBlockFactories as well, so they can propagate that to their children when we're explicitly instantiating the ListBlockFactory. Any thoughts on how to approach this?